src/components/organisms/NewProfileModal/NewProfileModal.tsx

```diff
@@ -29,102 +25,83 @@ interface NewProfileModalProps {
 }
 
 export const NewProfileModal: React.FC<NewProfileModalProps> = ({ venue }) => {
-  const { selectRecipientChat } = useChatSidebarControls();
-
-  const firebase = useFirebase();
-  const history = useHistory();
+  const {
+    selectedUserId,
+    hasSelectedProfile,
+    closeUserProfileModal,
+  } = useProfileModalControls();
 
   const {
-    isShown: editMode,
-    show: turnOnEditMode,
-    hide: turnOffEditMode,
-  } = useShowHide();
+    isShown: isModalShown,
+    hide: hideModal,
+    show: showModal,
+  } = useShowHide(true);
 
   const {
     isShown: isSubmitting,
     show: startSubmitting,
     hide: stopSubmitting,
   } = useShowHide();
 
-  const {
-    selectedUserProfile,
-    hasSelectedProfile,
-    closeUserProfileModal,
-  } = useProfileModalControls();
-
-  const isSameUser = useIsCurrentUser(selectedUserProfile?.id);
-
-  const openChosenUserChat = useCallback(() => {
-    if (!selectedUserProfile?.id) return;
-
-    selectRecipientChat(selectedUserProfile?.id);
-    closeUserProfileModal();
-  }, [selectRecipientChat, closeUserProfileModal, selectedUserProfile?.id]);
-
-  const logout = useCallback(async () => {
-    await firebase.auth().signOut();
-
-    history.push(
-      //@debt seems like a legacy logic
-      IS_BURN ? "/enter" : venue.id ? venueLandingUrl(venue.id) : "/"
-    );
-  }, [firebase, history, venue.id]);
-
   const hideHandler = useCallback(async () => {
     if (isSubmitting) return;
 
-    closeUserProfileModal();
-    setTimeout(
-      () => turnOffEditMode(),
-      PROFILE_MODAL_EDIT_MODE_TURNING_OFF_DELAY
-    );
-  }, [closeUserProfileModal, isSubmitting, turnOffEditMode]);
+    hideModal();
+    // when closeUserProfileModal is called modal hide animation starts
+    // but selectedUserId is immediately undefined and while the modal is sliding away
+    // an error message from ProfileModalFetchUser is shown
+
+    // This is to fix that.
+    setTimeout(() => {
+      closeUserProfileModal();
+      showModal();
+    }, REACT_BOOTSTRAP_MODAL_HIDE_DURATION);
+  }, [closeUserProfileModal, hideModal, isSubmitting, showModal]);
 
   const handleSubmitWrapper: (
     inner: OnSubmit<UserProfileModalFormData>
-  ) => OnSubmit<UserProfileModalFormData> = (
-    inner: OnSubmit<UserProfileModalFormData>
-  ) => async (data) => {
-    startSubmitting();
-    try {
-      await inner(data);
-    } finally {
-      stopSubmitting();
-    }
-  };
+  ) => OnSubmit<UserProfileModalFormData> = useCallback(
+    (inner: OnSubmit<UserProfileModalFormData>) => async (data) => {
+      startSubmitting();
+      try {
+        await inner(data);
+      } finally {
+        stopSubmitting();
+      }
+    },
+    [startSubmitting, stopSubmitting]
+  );
```

Comment:
It still is not using `useAsync`.

Reply:
It's maybe hard to spot but it's a function returning a function. And only the inner one is `async` so I cannot see how I can use `useAsync` here.

Reply:
You have useAsyncFn for this